### PR TITLE
Dual Antag Gamemodes, Calm Variants of game rules, and changeling delay matches traitor

### PR DESCRIPTION
**Pull request by TooSillyFennec**
## About the PR
Title :3

## Why / Balance
Changeling Changes - Same reason as agents, and also better for dual antag games for them to match. And changing the delay only for the Calm variants used in dual antag games would blatantly get meta'd. Also, only 5 can roll now- not that we really _get_ the pop for 6, but 6 would be carnage.

Calm Variants - Extra Player ratio variants that need 30+ Ready. Needed for dual antags to actually be balanced, but also an admin tool. All At once, Calm, no zombies anyone??? - This is sarcasm, all at once sucks :3

Weight Changes - Traitor 40%, Ling 10%, Traitorling 5%, ops 20%, optraitor and opling respectively get 1% each, revs 10%, revtraitor gets 2%, revling 1%. Rest stays the same. These can be changed mattering on how the experience is. Also, thanks piras for these more clean weights.

Also, some key things: - Annndd these are outdated already. Just check the "piras weights" commit and review.

50% of games include Traitors with 30+ Ready

28% of games include Changelings with 30+ Ready.

67% of games with 30+ Ready don't have major antags- AKA Revs, Zombs or ops. 

under 30+ Ready due to how the game works, all dual antag games become traitor whether they had them or not, which really is a good thing at that pop level, as lower pop's better with less major antags, and changeling's honestly less fun with only 1-2 Changelings and actually is more destructive at lower pops as they commonly have a pretty lacking sec/command count.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- add: Added gamemodes to secret with two antags at once, but have less of one type and need 30+ Players ready to start. 
- tweak: Weights have been slightly tweaked. Lings are rarer, and ops/revs are more common due to dual antag gamemodes.
- tweak: Changeling now rolls with the same delay as traitors, and changeling max has been lessened to 5.

### CODE REVIEW

**Comment by a-person5660** [comment]
Coated PR

**Comment by whateverusername0** [comment]
the duality of antag

**Comment by whateverusername0** [comment]
love these secret weights
now what if we add enother orundstart antag like cult, ratvar cult, heretics and other shit?
how many gamemodes do you wish to add? :trollface: 
srsly we need dynamic

**Comment by a-person5660** [comment]
I'm 90% sure the check fail isn't because of this

**Comment by whateverusername0** [comment]
mfw air consistency test fails
who farted

**Comment by a-person5660** [comment]
> srsly we need dynamic

That's a job for wizden :trollface:

**Comment by TooSillyFennec** [comment]
> love these secret weights now what if we add enother orundstart antag like cult, ratvar cult, heretics and other shit? how many gamemodes do you wish to add? :trollface: srsly we need dynamic

Pretty correct, but at the current antag level it's not needed **yet.**

By the time wizards are done and piras finishes cult? we really do need dynamic to make antag chances for certain antags not fucking once in a lifetime and allow extra antags to be added after. 0.05 Revs was already personally at least hell, imagine 0.04 wizard and then adding an extra antag.

Yeah, probably good to make a dynamic issue when we have at least one of those.

File: Resources/Prototypes/Goobstation/GameRules/roundstart.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  id: CalmRevs # For Dual Antag Gamemodes
+  parent: BaseGameRule
+  components:
+  - type: GameRule
+    minPlayers: 30
+  - type: RevolutionaryRule
+  - type: AntagSelection
+    definitions:
+    - prefRoles: [ HeadRev ]
+      max: 2
```
Comment (Piras314) [inline Resources/Prototypes/Goobstation/GameRules/roundstart.yml:137]: max should be 1

File: Resources/Prototypes/Goobstation/GameRules/roundstart.yml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  - type: GameRule
+    minPlayers: 15
+    delay:
+      min: 240
+      max: 420
+  - type: AntagSelection
+    agentName: changeling-roundend-name
+    definitions:
+    - prefRoles: [ Changeling ]
+      max: 6
```
Comment (Piras314) [inline Resources/Prototypes/Goobstation/GameRules/roundstart.yml:15]: out of the scope of this PR maybe but this should be 3 or 4

File: Resources/Prototypes/Goobstation/game_presets.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    - nuketot
+    - optraitor
+    - optot
+  name: nukeops-title
+  description: nukeops-description
+  showInVote: false
+  rules:
+    - Calmops
+    - CalmTraitor
+    - SubGamemodesRule
```
Comment (Piras314) [inline Resources/Prototypes/Goobstation/game_presets.yml:45]: no subgamemodes (thieves)

File: Resources/Prototypes/Goobstation/game_presets.yml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    - revtraitor
+    - revtot
+    - totrevs
+  name: revtraitor-title
+  description: revtraitor-description
+  showInVote: false
+  rules:
+    - CalmRevs
+    - CalmTraitor
+    - SubGamemodesRule
```
Comment (Piras314) [inline Resources/Prototypes/Goobstation/game_presets.yml:78]: no subgamemodes (thieves)

File: Resources/Prototypes/Goobstation/game_presets.yml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  alias:
+    - revling
+    - lingrevs
+  name: revling-title
+  description: revling-description
+  showInVote: false
+  rules:
+    - CalmRevs
+    - CalmLing
+    - SubGamemodesRule
```
Comment (Piras314) [inline Resources/Prototypes/Goobstation/game_presets.yml:93]: no subgamemodes (thieves)

**Review by Piras314 — CHANGES_REQUESTED** [review]
the weight numbers hurt my eyes

File: Resources/Prototypes/secret_weights.yml [hunk 6]
```diff
@@ -1,10 +1,16 @@
...
   id: Secret
   weights:
-    Nukeops: 0.20
-    Traitor: 0.40
-    Changeling: 0.20 # Goobstation
-    Zombie: 0.05
-    Survival: 0.05 # Goobstation
-    Revolutionary: 0.10 # Goobstation
-  # Wizard: 0.05
+    Traitor: 0.38 # Goobstation
```
[the remark this answers is not in the document]
Comment (Piras314) [reply Resources/Prototypes/secret_weights.yml:4]: rethinking this

[the remark this answers is not in the document]
Comment (Piras314) [reply Resources/Prototypes/secret_weights.yml:4]: traitor 40 (total 43)
cling 10 (total 17)
tatorling 5
nukeops 20 (total 22)
nuketraitor 1
nukeling 1
revolutionary 10 (total 13)
revtraitor 2
revling 1
zombie 5
survival 5

[the remark this answers is not in the document]
Comment (a-person5660) [reply Resources/Prototypes/secret_weights.yml]: Zombie chance is too high imo, it's the worst mode, should be 1%.

File: Resources/Locale/en-US/Goobstation/game-ticking/game-presets/presets-dualantag.ftl [hunk 7]
```diff
@@ -2,7 +2,7 @@ traitorling-title = Traitorling
 traitorling-description = Attention. Known enemy signals and strange biosigns detected. Confirmed Syndicate Agents and Changelings on board.
 
 revtraitor-title = Revolutionary Traitors
-revtraitor-description = A revolution have been provoked by a member of the Syndicate, but not every Syndicate member got the hint...
+revtraitor-description = A revolution have been provoked by a member of the Syndicate, but not every agent got the hint...
```
Comment (Piras314) [inline Resources/Locale/en-US/Goobstation/game-ticking/game-presets/presets-dualantag.ftl:5]: I know this is minor but the main point of my correction was you said "have been provoked" when it should be "has been provoked" @TooSillyFennec other than that this is good I think

**Review by Piras314 — CHANGES_REQUESTED** [review]

File: Resources/Locale/en-US/Goobstation/game-ticking/game-presets/presets-dualantag.ftl (hunk 7 above)
Comment (TooSillyFennec) [reply Resources/Locale/en-US/Goobstation/game-ticking/game-presets/presets-dualantag.ftl:5]: can fix that

**Review by Piras314 — APPROVED** [review]

**Pull request merged**